pkg/agent/cniserver/ipam/antrea_ipam.go

```diff
@@ -192,6 +194,89 @@ func (d *AntreaIPAM) Check(args *invoke.Args, k8sArgs *argtypes.K8sArgs, network
 	return true, nil
 }
 
+func (d *AntreaIPAM) secondaryNetworkAdd(args *invoke.Args, k8sArgs *types.K8sArgs, networkConfig *types.NetworkConfig) (*current.Result, error) {
+	ipamConf := networkConfig.IPAM
+	if len(ipamConf.IPPools) == 0 {
+		return nil, fmt.Errorf("Antrea IPPool must be specified")
+	}
+
+	if err := d.waitForControllerReady(); err != nil {
+		// Return error to let the invoker retry.
+		return nil, err
+	}
+
+	var err error
+	var allocator *poolallocator.IPPoolAllocator
+	for _, p := range ipamConf.IPPools {
+		allocator, err = d.controller.getPoolAllocatorByName(p)
+		if err == nil {
+			// Use the first IPPool that exists.
+			break
+		}
+		if !errors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get IPPool %s: %v", p, err)
+		}
+		klog.InfoS("IPPool not found", "pool", p)
+
+	}
+	if allocator == nil {
+		return nil, fmt.Errorf("no valid IPPool found")
+	}
+
+	owner := getAllocationOwner(args, k8sArgs, nil, true)
+	var ip net.IP
+	var subnetInfo *crdv1a2.SubnetInfo
+	ip, subnetInfo, err = allocator.AllocateNext(crdv1a2.IPAddressPhaseAllocated, owner)
+	if err != nil {
+		return nil, err
+	}
+
+	// Copy routes and DNS from the input IPAM configuration.
+	result := &current.Result{Routes: ipamConf.Routes, DNS: ipamConf.DNS}
+	gwIP := net.ParseIP(subnetInfo.Gateway)
+	ipConfig, _ := generateIPConfig(ip, int(subnetInfo.PrefixLength), gwIP)
+	result.IPs = append(result.IPs, ipConfig)
+	return result, nil
+}
+
+func (d *AntreaIPAM) secondaryNetworkDel(args *invoke.Args, k8sArgs *types.K8sArgs, networkConfig *types.NetworkConfig) error {
+	owner := getAllocationOwner(args, k8sArgs, nil, true)
+	_, err := d.del(owner.Pod)
+	return err
+}
+
+func (d *AntreaIPAM) secondaryNetworkCheck(args *invoke.Args, k8sArgs *types.K8sArgs, networkConfig *types.NetworkConfig) error {
+	return fmt.Errorf("CNI CHECK is not implemented for secondary network")
+}
+
+func (d *AntreaIPAM) del(podOwner *crdv1a2.PodOwner) (foundAllocation bool, err error) {
+	if err := d.waitForControllerReady(); err != nil {
+		// Return error to let the invoker retry.
+		return false, err
+	}
+	// The Pod resource might have been removed; and for a secondary we
+	// would rely on the passed IPPool for CNI DEL. So, search IPPools with
+	// the matched PodOwner.
+	allocators, err := d.controller.getPoolAllocatorsByOwner(podOwner)
+	if err != nil {
+		return false, err
+	}
+
+	if len(allocators) == 0 {
+		return false, nil
+	}
+	// Multiple allocators can be returned if the network interface has IPs
+	// allocated from more than one IPPools.
+	for _, a := range allocators {
+		err = a.ReleaseContainer(podOwner.ContainerID, podOwner.IFName)
+		if err != nil {
+			klog.Errorf("xxx err: %v", err)
```

Comment:
Looks like forgotten debug print

Reply:
Yes, you are right. I found that too (together with another minor issue), but unfortunately after merging the PR :(

Fixed it in #3606